packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -185,3 +189,280 @@ class Xcode {
+       _logger = logger,
+       _xcode = xcode;
+
+  final ProcessUtils _processUtils;
+  final Logger _logger;
+  final Xcode _xcode;
+
+  bool get isInstalled => _xcode.isInstalledAndMeetsVersionCheck && xcdevicePath != null;
+
+  String _xcdevicePath;
+  String get xcdevicePath {
+    if (_xcdevicePath == null) {
+      try {
+        _xcdevicePath = _processUtils.runSync(
+          <String>[
+            'xcrun',
+            '--find',
+            'xcdevice'
+          ],
+        ).stdout.trim();
+      } on ProcessException {
+        // Ignored, return null below.
+      } on ArgumentError {
+        // Ignored, return null below.
+      }
+    }
+    return _xcdevicePath;
+  }
+
+  Future<List<dynamic>> _getAllDevices() async {
+    if (!isInstalled) {
+      return null;
+    }
+    try {
+      // USB-tethered devices should be found quickly. 1 second timeout is faster than the default.
+      final RunResult result = await _processUtils.run(
+        <String>[
+          xcdevicePath,
+          'list',
+          '--timeout',
+          '1',
+        ],
+      );
+      if (result.exitCode == 0) {
+        return json.decode(result.stdout) as List<dynamic>;
+      }
+      _logger.printTrace('xcdevice returned an error:\n${result.stderr}');
+    } on ProcessException {
+      _logger.printTrace('Failed to invoke xcdevice');
+    }
+
+    return null;
+  }
+
+  /// List of devices available over USB.
+  Future<List<IOSDevice>> getAvailableTetheredIOSDevices() async {
+    final List<dynamic> allAvailableDevices = await _getAllDevices();
+
+    if (allAvailableDevices == null) {
+      return const <IOSDevice>[];
+    }
+
+    // [
+    //  {
+    //    "simulator" : true,
+    //    "operatingSystemVersion" : "13.3 (17K446)",
+    //    "available" : true,
+    //    "platform" : "com.apple.platform.appletvsimulator",
+    //    "modelCode" : "AppleTV5,3",
+    //    "identifier" : "CBB5E1ED-2172-446E-B4E7-F2B5823DBBA6",
+    //    "architecture" : "x86_64",
+    //    "modelName" : "Apple TV",
+    //    "name" : "Apple TV"
+    //  },
+    //  {
+    //    "simulator" : false,
+    //    "operatingSystemVersion" : "13.3 (17C54)",
+    //    "interface" : "usb",
+    //    "available" : true,
+    //    "platform" : "com.apple.platform.iphoneos",
+    //    "modelCode" : "iPhone8,1",
+    //    "identifier" : "d83d5bc53967baa0ee18626ba87b6254b2ab5418",
+    //    "architecture" : "arm64",
+    //    "modelName" : "iPhone 6s",
+    //    "name" : "iPhone"
+    //  },
+    //  {
+    //    "simulator" : true,
+    //    "operatingSystemVersion" : "6.1.1 (17S445)",
+    //    "available" : true,
+    //    "platform" : "com.apple.platform.watchsimulator",
+    //    "modelCode" : "Watch5,4",
+    //    "identifier" : "2D74FB11-88A0-44D0-B81E-C0C142B1C94A",
+    //    "architecture" : "i386",
+    //    "modelName" : "Apple Watch Series 5 - 44mm",
+    //    "name" : "Apple Watch Series 5 - 44mm"
+    //  },
+    // ...
```

Comment:
This method is pretty long -- could a few smaller methods be factored out to break out? Something like some {sub,super,}set of:
* fetch blob o' json (already done: `_getAllDevices()`)
* `isIPhoneOSDevice(deviceProperties)`
* error message check (already done: `_parseErrorMessage`)
* `_isAvailable(...)` 
* `_isUSBDevice(...)` (or alternatively `_getConnectionType(...) == _DeviceConnectionType.usb`)
* get SDK version
* get cpu arch

Reply:
Good idea, done!